### PR TITLE
refactor(renterd): migrate to events APIs and remove other deprecated APIs

### PR DESCRIPTION
**Pull request by alexfreska**
> All checks pass on final api-breakers PR: https://github.com/SiaFoundation/web/pull/743

This PR corresponds to the changes in: https://github.com/SiaFoundation/renterd/pull/1478

### Updates
- Removed deprecated wallet transactions and pending transactions APIs.
- Added wallet events and pending events APIs.
- Removed deprecated wallet discard, fund, sign, and outputs APIs.
- The renterd transactions list is now includes more details such as specific transaction types and maturity height for locked siacoin.

### CODE REVIEW

**Comment by alexfreska** [comment]
* **#732** <a href="https://app.graphite.dev/github/pr/SiaFoundation/web/732?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alexfreska and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: .changeset/beige-cycles-change.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'renterd': minor
+---
+
+The transactions list is now includes more details such as specific transaction types and maturity height for locked siacoin.
```
Comment (peterjan) [inline .changeset/beige-cycles-change.md:5]:
```suggestion
The transactions list now includes more details such as specific transaction types and maturity height for locked siacoin.
```

File: apps/renterd/contexts/transactions/index.tsx [hunk 2]
```diff
@@ -66,52 +55,91 @@ function useTransactionsMain() {
     },
   })
 
-  const { openDialog } = useDialog()
-
-  const siascanUrl = useSiascanUrl()
+  const { filters, setFilter, removeFilter, removeLastFilter, resetFilters } =
+    useServerFilters()
```
Comment (peterjan) [inline apps/renterd/contexts/transactions/index.tsx:59]: return values aren't used

File: apps/renterd/contexts/transactions/types.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+export type TableColumnId =
+  | 'transactionId'
+  | 'type'
+  | 'height'
+  | 'timestamp'
+  | 'amount'
+  | 'fee'
+  | 'transactionId'
```
Comment (peterjan) [inline apps/renterd/contexts/transactions/types.ts:33]: `transactionId` is duplicate

File: apps/renterd/contexts/transactions/index.tsx (hunk 2 above)
Comment (alexfreska) [reply apps/renterd/contexts/transactions/index.tsx:59]: Removed, as we do not support any filtering yet.

**Review by ChrisSchinnerl — APPROVED** [review]

**Comment by alexfreska** [comment]
### Merge activity

* **Sep 17, 10:13 AM EDT**: @alexfreska started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/SiaFoundation/web/732).

**Pull request merged**